packages/client/test/rpc/helpers.ts

```diff
@@ -166,11 +166,12 @@ export async function baseRequest(
 export async function setupChain(genesisFile: any, chainName = 'dev', clientOpts: any = {}) {
   const genesisParams = await parseCustomParams(genesisFile, chainName)
   const genesisState = genesisFile.alloc ? await parseGenesisState(genesisFile) : {}
+
   const common = new Common({
     chain: chainName,
     customChains: [[genesisParams, genesisState]],
   })
-  common.setHardforkByBlockNumber(0)
+  common.setHardforkByBlockNumber(0, genesisParams.td)
```

Comment:
this would be the total terminal difficulty for the chain, not the difficulty for the genesis block, so you only need to do `genesisParams.difficulty`. please watch for #1741, as this will change to a number to better match up our libs, and then you can wrap with `new BN()` if needed. i wll rebase my kiln branch as well once merged.

Reply:
ok, have merged and updated merge-kiln branch

Reply:
Correct, that's the `totalTerminalDifficulty` - Got inspired from [this line](https://github.com/ethereumjs/ethereumjs-monorepo/blob/merge-kiln/packages/client/lib/sync/fullsync.ts#L61) that I mentioned before; if I got this right, we should pass this variable instead of `difficulty`. Note that for `post-merge` mock data the block difficulty is `0x400000000` and not `0`

That's why I thought that we needed to pass this variable, just for the sake of setting the hard fork

Please let me know if this does not make sense